### PR TITLE
Add new types to Calendar.ISO

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/calendar/iso.ex [hunk 1]
```diff
@@ -41,12 +41,35 @@ defmodule Calendar.ISO do
   @type year :: -9999..9999
   @type month :: 1..12
   @type day :: 1..31
+  @type hour :: 0..23
+  @type minute :: 0..59
+  @type second :: 0..59
+
+  @type day_of_era :: Calendar.day_of_era()
```
Comment (eksperimental) [inline lib/elixir/lib/calendar/iso.ex]: this type is added in this PR: https://github.com/elixir-lang/elixir/pull/9511
please do not merge until the other PR is merged, or modify this type

**Comment by josevalim** [comment]
Depending on `Calendar.type` is actually clearer IMO, because it is clear they are meant to be the same, and not some overwrite. Thanks.

File: lib/elixir/lib/calendar/iso.ex [hunk 2]
```diff
@@ -736,7 +759,7 @@ defmodule Calendar.ISO do
 
   """
   @doc since: "1.8.0"
-  @spec day_of_era(year, month, day) :: {day, era}
+  @spec day_of_era(year, month, day) :: day_of_era()
```
Comment (eksperimental) [inline lib/elixir/lib/calendar/iso.ex]: this is a bug fix

**Comment by eksperimental** [comment]
@josevalim I can remove the changes that remove the `Calendar` module. but there are other changes that are positive as having the types hour/minute/second, and there's a bug fix. Could you please reopen the PR so I can submit those changes

**Comment by josevalim** [comment]
Reopen! We should generally avoid mixing bug fixes with refactoring. :) thanks!

**Comment by eksperimental** [comment]
Please have a look again

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**